sympy/discrete/transforms.py

```diff
@@ -0,0 +1,68 @@
+"""
+Discrete Fourier Transform, Walsh Hadamard Transform,
+Number Theoretic Transform, Zeta Transform, Mobius Transform
+"""
+from __future__ import print_function, division
+
+from sympy.core import S, Symbol, sympify
+from sympy.core.compatibility import as_int, range
+from sympy.core.function import expand, expand_mul
+from sympy.core.numbers import pi, I
+from sympy.functions.elementary.exponential import exp
+from sympy.ntheory import isprime, primitive_root
+from sympy.utilities.iterables import ibin
+
+def _fourier_transform(seq, inverse=False):
+    """
+    Performs the DFT in complex field, by padding zeros as
+    radix 2 FFT, requires the number of sample points to be
+    a power of 2.
+    """
+
+    if not hasattr(seq, '__iter__'):
+        raise TypeError("Expected a sequence of numeric coefficients " +
+                        "for Fourier Transform")
+
+    a = sympify(seq)
```

Comment:
This could be made a list to begin with: `a = [sympify(arg) for arg in seq]`. (`a = list(map(sympify, seq))` is also often used.)